Composer/packages/server/src/models/bot/builder.ts

```diff
@@ -98,6 +102,59 @@ export class Builder {
     this._locale = v;
   }
 
+  /**
+   * Orchestrator: Get available list of NLR models
+   */
+  public async runOrchestratorNlrList(): Promise<IOrchestratorNLRList> {
+    return JSON.parse(await Orchestrator.nlrListAsync());
+  }
+
+  /**
+   * Orchestrator: Download an available NLR model.
+   *
+   * @remarks Available NLR models and VersionIds are obtained by running runOrchestratorNlrList first.
+   *
+   * @param modelPath - Folder path to save NLR model
+   * @param nlrId - VersionId of the model
+   * @param onProgress - Callback to notify of D/L progress
+   * @param onFinish - Callback to notify of D/L completed
+   */
+  public async runOrchestratorNlrGet(
+    modelPath: string,
+    nlrId: string,
+    onProgress: IOrchestratorProgress,
+    onFinish: IOrchestratorProgress
+  ): Promise<void> {
+    await Orchestrator.nlrGetAsync(modelPath, nlrId, onProgress, onFinish);
+  }
+
+  /**
+   * Orchestrator: Build command to compile .lu files into Binary LU (.blu) snapshots.
+   *
+   * A snapshot (.blu file) is created per .lu supplied
+   *
+   * @param files - Array of FileInfo
+   * @param modelPath - Path to NLR model folder
+   * @param isDialog - Flag to toggle creation of Recognizer Dialogs (default: true)
+   * @param fullEmbedding - Use larger embeddings and skip size optimization (default: false)
+   * @returns An object containing snapshot bytes and recognizer dialogs for each .lu file
+   */
+  public async runOrchestratorBuild(
+    files: FileInfo[],
+    modelPath: string,
+    isDialog = true,
+    fullEmbedding = false
+  ): Promise<IOrchestratorBuildOutput> {
+    const luObjects = files
+      .filter((fi) => fi.name.endsWith('.lu'))
```

Comment:
>.lu [](start = 40, length = 3)

Same here. Maybe we also need to filter .qna files

Reply:
Do we need to filter the files with the recognizer type now? The build function parameter luFiles is referred by CrossTrainedRecognizer and LuisRecognizer. Maybe we can separate the files by recognizer type in the future.

Reply:
I think we will filter it later maybe in the caller side of this function. Not in here. This PR's focus should be bring the lib in, and with proper test case to verify the lib is working as expected.